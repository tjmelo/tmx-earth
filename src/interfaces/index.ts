import { ReactNode } from "react"

export interface ITargetEvent {
    value: string | null
}

export interface ICommonName {
    name: { common: string }
    flag: string
}

export interface ILoading {
    type: string,
    children: string
}

export interface ICountry {
    country: { country: string }
}

export interface IMountListCountries {
    data: any
}
export interface ISectionList{
    data: ReactNode | any,
    subdata?: ReactNode,
    children: string | ReactNode
}
