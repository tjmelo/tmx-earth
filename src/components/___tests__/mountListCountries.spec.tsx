import {render, screen} from '@testing-library/react'
import '@testing-library/jest-dom'

import MountListCountries from '../Mount'
import { TListData } from '../../interfaces'

const dataTest: TListData = {
    flags: {
        svg: 'SVG data test'
    },
    name: {
        official: 'Test data',
        nativeName: {
            'test': {
                common: 'Common name test'
            }
        },
        common: ''
    },
    coatOfArms: {
        svg: 'SVG data test',
    },
    languages: [
        'language test',
        'other language test'
    ],
    population: 101010,
    area: 999999,
    currencies: {},
    capital: '',
    region: '',
    subregion: '',
    borders: [],
    tld: []
}

test('Should render a Mount List Country', () => {
    //do
    const {asFragment} = render(<MountListCountries data={dataTest}/>)

    // then
    expect(screen.getByText('Common name test')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot();
})