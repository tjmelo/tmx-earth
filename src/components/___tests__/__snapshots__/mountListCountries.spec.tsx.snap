// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a Mount List Country 1`] = `
<DocumentFragment>
  <div
    class="row my-5"
  >
    <div
      class="col-xs-12 d-flex mb-2 flag"
    >
      <figure>
        <img
          alt="Official name test"
          src="SVG data test"
        />
      </figure>
      <span
        class="text-primary name"
      >
        Common name test
      </span>
      <img
        alt="Official name test"
        height="30"
        src="SVG data test"
        width="30"
      />
    </div>
    <div
      class="col-xs-12 my-3 border-bottom"
    >
      <span
        class="text-secondary"
      >
        Native name:
      </span>
      <br />
      <strong
        class="text-primary"
      />
      <br />
    </div>
    <div
      class="col-xs-12 my-3 border-bottom"
    >
      <span
        class="text-secondary"
      >
        Capital:
      </span>
      <br />
      <strong
        class="text-primary"
      >
        -----
      </strong>
      <br />
    </div>
    <div
      class="col-xs-12 my-3 border-bottom"
    >
      <span
        class="text-secondary"
      >
        Region:
      </span>
      <br />
      <strong
        class="text-primary"
      >
        -----
      </strong>
      <br />
    </div>
    <div
      class="col-xs-12 my-3 border-bottom"
    >
      <span
        class="text-secondary"
      >
        Subregion:
      </span>
      <br />
      <strong
        class="text-primary"
      >
        -----
      </strong>
      <br />
    </div>
    <div
      class="col-xs-12 my-3 border-bottom"
    >
      <span
        class="text-secondary"
      >
        Languages:
      </span>
      <br />
      <strong
        class="text-primary"
      >
        language test
      </strong>
      <br />
    </div>
    <div
      class="col-xs-12 my-3 border-bottom"
    >
      <span
        class="text-secondary"
      >
        Currencies:
      </span>
      <br />
      <strong
        class="text-primary"
      />
      <br />
    </div>
    <div
      class="col-xs-12 col-sm-6 col-md-4 col-lg-4 mb-3 border-bottom"
    >
      <span
        class="text-secondary"
      >
        Borders: 
      </span>
       
      <br />
      <strong
        class="text-primary"
      >
        -----
      </strong>
    </div>
    <div
      class="col-xs-12 my-3 border-bottom"
    >
      <span
        class="text-secondary"
      >
        Poulation:
      </span>
      <br />
      <strong
        class="text-primary"
      >
        101.010
      </strong>
      <br />
    </div>
    <div
      class="col-xs-12 my-3 border-bottom"
    >
      <span
        class="text-secondary"
      >
        Area:
      </span>
      <br />
      <strong
        class="text-primary"
      >
        999.999
      </strong>
      <br />
    </div>
    <div
      class="col-xs-12 my-3 border-bottom"
    >
      <span
        class="text-secondary"
      >
        Domain:
      </span>
      <br />
      <strong
        class="text-primary"
      >
        -----
      </strong>
      <br />
    </div>
  </div>
</DocumentFragment>
`;
